src/lightning/pytorch/plugins/precision/fsdp.py

```diff
@@ -88,10 +125,86 @@ def init_context(self) -> Generator[None, None, None]:
         torch.set_default_dtype(default_dtype)
 
     @contextmanager
-    def forward_context(self) -> Generator[None, None, None]:
-        """For FSDP, this context manager is a no-op since conversion is already handled internally.
+    def forward_context(self) -> Generator:
+        if "mixed" in self.precision:
+            with self._autocast_context_manager():
+                yield
+        else:
+            default_dtype = torch.get_default_dtype()
+            torch.set_default_dtype(self._desired_input_dtype)
+            yield
+            torch.set_default_dtype(default_dtype)
+
+    def convert_input(self, data: Any) -> Any:
+        return apply_to_collection(data, function=_convert_fp_tensor, dtype=Tensor, dst_type=self._desired_input_dtype)
+
+    def convert_output(self, data: Any) -> Any:
+        return apply_to_collection(data, function=_convert_fp_tensor, dtype=Tensor, dst_type=torch.get_default_dtype())
```

Comment:
shall these rather protected?

Reply:
They are part of the base Precision class: https://github.com/Lightning-AI/lightning/blob/70e31b64800a89e91ff46b631670cf8944d10ab9/src/lightning/fabric/plugins/precision/precision.py#L63-L72
and here we are simply overriding them, and so they need to be public here :)